README.md

```diff
@@ -13,312 +13,252 @@ to [dev.poktroll.com/category/actors](https://dev.poktroll.com/category/actors).
-}
+Refer to [block.go](https://github.com/pokt-network/shannon-sdk/blob/main/block.go)
+for detailed information.
 
-endpoints := sdk.Endpoints(session, serviceID)
-```
+The default implementation uses the `CosmosSDK`'s `http.HTTP` client to fetch the
+block height.
 
-#### Public vs Private Fields
+#### Signer
 
-The goal of this `SDK` is to make all fields of concrete types public to the user
-if there is a potential need for the user to set them directly.
+The `Signer` signs `RelayRequests` to ensure their authenticity and integrity.
 
-**IT SHOULD** be possible for the user to initialize any component of the SDK by
-creating a struct and setting the bare minimum necessary fields.
+It provides the following method:
 
-For example, the SDK biases towards the following design:
+- `Sign`: Signs a given `RelayRequest` using the provided `ApplicationRing`.
 
-```go
-c := SessionClient {
-     HttpClient:  myCustomHttpTransport
-}
-```
+The `Signer` must set its `PrivateKeyHex` field to the private key of the associated
+application or gateway.
 
-Instead of the following design:
+Refer to [signer.go](https://github.com/pokt-network/shannon-sdk/blob/main/signer.go)
+for detailed information.
 
-```go
-c := NewSessionClient(nil, nil, myCustomHttpTransport, nil, nil)
-```
+#### Session Client
+
+The `SessionClient` retrieves `Session` information from the Pocket network for a specified `Application` address, `Service.Id`, and block height.
+
+It provides the following method:
+
+- `GetSession`: Retrieves session information for a given `Application` address,
+`Service.Id`, and block height.
+
+
+The `SessionClient` relies on the `PoktNodeSessionFetcher` interface, which requires implementations to fetch session information from the Pocket network.
+
+Refer to [session.go](https://github.com/pokt-network/shannon-sdk/blob/main/session.go)
+for detailed information.
+
+#### Filtered Session
+
+To select the best supplier endpoint among available options, the SDK offers a
+`FilteredSession` struct that filters out endpoints that do not meet specified criteria.
+
+The `FilteredSession` requires an assigned `Session` in its `Session` field.
+It supports adding filter functions to its `EndpointFilters` field to select the
+best supplier endpoint.
+
+`FilteredSession` provides the following methods:
+
+- `AllEndpoints`: Retrieves all `Endpoints` from the `Session`, mapped to their
+respective `Supplier` addresses, allowing retrieval of all available supplier
+endpoints and performing custom filtering.
+- `FilteredEndpoints`: Retrieves filtered endpoints based on specified filter functions.
+Returned endpoints must pass all filter functions to be considered valid.
+
+Filtered endpoints adhere to the `Endpoint` interface, which provides:
+
+- `Header`: Retrieves the `Session` header corresponding to the `Supplier`'s endpoint.
+- `Supplier`: Retrieves the `Supplier` address corresponding to the `Endpoint`.
+- `Endpoint`: Retrieves the `url.URL` of the endpoint.
```

Comment:
I feel like it would be more consistent if these all had a `Get` prefix. Wdyt about leaving a comment and/or opening an issue for this?